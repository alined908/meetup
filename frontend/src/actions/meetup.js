import {GET_MEETUPS_REQUEST, GET_MEETUPS_SUCCESS, GET_MEETUPS_ERROR, 
    ADD_MEETUP, ADD_MEETUP_MEMBER, ADD_EVENT_OPTION, ADD_GLOBAL_MESSAGE, DELETE_MEETUP, EDIT_MEETUP,
    VOTE_MEETUP_EVENT, GET_MEETUP_EVENTS, DELETE_MEETUP_EVENT, ADD_MEETUP_EVENT, EDIT_MEETUP_EVENT} from "../constants/action-types";
import {axiosClient} from '../accounts/axiosClient';
import {history} from '../components/MeetupApp'

export const getMeetups = (data) => async dispatch => {
    dispatch({type: GET_MEETUPS_REQUEST})
    var params;
    if (data.type === "private"){
        params = {type: data.type, ...data.categories && {categories: data.categories}}
    } else {
        params = {
            type: data.type, ...data.categories && {categories: data.categories}, 
            latitude: data.coords.latitude, longitude: data.coords.longitude, 
            radius: data.coords.radius
        }
    }
    
    try {
        const response = await axiosClient.request({
            method: "GET",
            url: "/api/meetups/", 
            headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            },
            params: params
        })
        setTimeout(() => dispatch({type: GET_MEETUPS_SUCCESS, payload: response.data.meetups}), 100)
    } catch(e){
        console.log(e)
        dispatch({type: GET_MEETUPS_ERROR, payload: e})
    }
}

export const getMeetup = (uri) => async dispatch => {
    try {
        const response = await axiosClient.get(
            `/api/meetups/${uri}/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            }}
        )
        // console.log(response)
        dispatch({type: ADD_MEETUP, payload: response.data})
    } catch(e){
        history.push('/404')
    }
}

export const addMeetup = (formProps, redirectOnSuccess) => async dispatch => {
    var params = {...formProps, 
        date: formProps.date.getFullYear()+ "-" + ("0"+(formProps.date.getMonth()+1)).slice(-2) + "-" + ("0" + formProps.date.getDate()).slice(-2)
    }
    try {
        const response = await axiosClient.post(
            `/api/meetups/`, params, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: ADD_MEETUP, payload: response.data.meetup})
        console.log(response.data)
        redirectOnSuccess(response.data.meetup.uri)
    } catch(e){
        console.log(e)
    }
} 

export const editMeetup = (formProps, uri, redirectOnSuccess) => async dispatch => {
    console.log('edit meetup')
    var date = new Date(formProps.date)
    var params = {...formProps, 
        date: date.getFullYear()+ "-" + ("0"+(date.getMonth()+1)).slice(-2) + "-" + ("0" + date.getDate()).slice(-2)
    }
    try {
        const response = await axiosClient.patch(
            `/api/meetups/${uri}/`, params, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: EDIT_MEETUP, payload: response.data})
        console.log(response.data)
        redirectOnSuccess(uri)
    } catch(e){
        console.log(e)
    }
}

export const deleteMeetup = (uri) => async dispatch => {
    try {
        const response = await axiosClient.delete(
            `/api/meetups/${uri}/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        console.log(response)
        dispatch({type: DELETE_MEETUP, payload: uri})
        history.push(`/meetups`)
    } catch(e) {
        console.log(e)
    }
}

export const getMeetupEvents = (uri) => async dispatch => {
    try {
        const response = await axiosClient.get(
            `/api/meetups/${uri}/events/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        console.log(response)
        dispatch({type: GET_MEETUP_EVENTS, payload: {uri: uri, data: response.data}})
    } catch(e) {
        console.log(e)
    }
}

export const addMeetupEvent = (event) => async dispatch => {
    dispatch({type: ADD_MEETUP_EVENT, payload: event.message})
}

export const deleteMeetupEvent = (event) => async dispatch => {
    dispatch({type: DELETE_MEETUP_EVENT, payload: event.message})
}

export const reloadMeetupEvent = (event) => async dispatch => {
    dispatch({type: EDIT_MEETUP_EVENT, payload: event.message})
}

export const voteMeetupEvent = (event) => async dispatch => {
    dispatch({type: VOTE_MEETUP_EVENT, payload: event.message})
}
    
export const decideMeetupEvent = (event) => async dispatch => {
    dispatch({type: EDIT_MEETUP_EVENT, payload: event.message})
}

export const addMeetupMember = (event) => async dispatch => {
    dispatch({type: ADD_MEETUP_MEMBER, payload: event.message})
}

export const addEventOption = (event) => async dispatch => {
    dispatch({type: ADD_EVENT_OPTION, payload: event.message})
}

export const sendMeetupEmails = (uri) => async dispatch => {

    try {
        await axiosClient.post(
            `/api/meetups/${uri}/email/`, {}, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: ADD_GLOBAL_MESSAGE, payload: {type: "success", message: "Successfully sent emails"}})
    } catch(e) {
        dispatch({type: ADD_GLOBAL_MESSAGE, payload: {type: "error", message: "Unable to send emails"}})
    }
}