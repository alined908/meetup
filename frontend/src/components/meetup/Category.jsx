import React from 'react'
import {Button} from "@material-ui/core"

const Category = (props) => {
    const [variant, setVariant] = React.useState(props.clicked)

    const handleClick = () => {
        setVariant(variant === 'outlined' ? 'contained' : 'outlined')
    }

    return (
        <Button variant={variant} color="primary" onClick={() => handleClick()}>{props.category.label}</Button>
    )
}

export default Category