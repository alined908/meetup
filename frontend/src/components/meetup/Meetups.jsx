import React, {Component} from 'react';
import {connect} from 'react-redux';
import {getMeetups, getPreferences} from "../../actions";
import {MeetupCard, CategoryAutocomplete, MeetupForm} from "../components"
import {Grid, Grow, Tooltip, IconButton, Avatar,  CircularProgress, Paper} from '@material-ui/core'
import {Link} from 'react-router-dom'
import moment from "moment"
import {Public as PublicIcon, Add as AddIcon, Search as SearchIcon, Edit as EditIcon, Error as ErrorIcon, People as PeopleIcon} from '@material-ui/icons'
import PropTypes from "prop-types"
import { userPropType, preferencePropType, meetupPropType} from '../../constants/prop-types';
import {Helmet} from "react-helmet";
import 'react-dates/lib/css/_datepicker.css';
import 'react-dates/initialize';
import { DateRangePicker, isInclusivelyAfterDay } from "react-dates";
import styles from '../../styles/meetup.module.css'

class Meetups extends Component {
    constructor(props){
        super(props)
        this.state = {
            focusedInput: null,
            startDate: moment(),
            endDate: moment().add('7', 'd'),
            public: true,
            newMeetupForm: false,
            entries: [],
            preferences: [],
            clickedPreferences: []
        }
    }
    
    async componentDidMount(){
        await Promise.all([
            //this.props.getMeetups(),
            this.props.getMeetups({
                type: "public",
                startDate: this.state.startDate.format("YYYY-MM-DD"),
                endDate: this.state.endDate.format("YYYY-MM-DD"),
                categories: this.formatCategories([]), 
                coords: {
                    ...this.props.user.settings
                }
            }),
            this.props.getPreferences(this.props.user.id)
        ])
    }

    componentDidUpdate(){
        if (this.props.preferences !== this.state.preferences){
            this.setState({
                preferences: this.props.preferences,
            })
        }
    }

    /**
     * Get public meetups or private meetups
     * @param {string} type - Public or Private
     */
    handleMeetupsType = (type) => {
        let publicBool = this.state.public;
        if (type === "public") {
            if (!publicBool){
                publicBool = true
            }
        } else if (type === "private"){
            if (publicBool){
                publicBool = false
            }
        }
        this.setState({public: publicBool}, () => this.determineGetMeetups(publicBool, this.state.entries))
    }


    /**
     * Filter public/private meetups by preferences.
     * @param {number} index - Index of clicked preference
     */
    handlePreferenceClick = (index) => {
        var category = this.state.preferences[index].category
        const clickedPreferences = [...this.state.clickedPreferences]
        var entries;
        if (!clickedPreferences[index]){
            entries = [...this.state.entries, category]
        } else {
            entries = this.state.entries.filter(entry => JSON.stringify(entry) !== JSON.stringify(category))
        }
        clickedPreferences[index] = !clickedPreferences[index]
        this.setState({clickedPreferences, entries}, () => this.determineGetMeetups(this.state.public, entries))
    }

    /**
     * Filter public/private meetups by date range.
     * @param {Object} - Start date - End date
     */
    onDatesChange = ({startDate, endDate}) => {
        if (!startDate || !endDate) return;
        if (!startDate.isValid() || !endDate.isValid()) return;
        this.setState({startDate, endDate}, () => this.determineGetMeetups(this.state.public, this.state.entries))
    }

    onFocusChange = (focusedInput) => {
        this.setState({focusedInput})
    }

    isOutsideRange = day => {
        return !isInclusivelyAfterDay(day, moment()) || day.isAfter(moment().add('30', 'd'))
    }

    /**
     * Filter public/private meetups by category from autocomplete.
     * @param {Array} values - Array of categories.
     */
    onTagsChange = (event, values) => {
        console.log(values)
        var clickedPrefs = [...this.state.clickedPreferences]
  
        for(var i = 0; i < clickedPrefs.length; i ++){
            if (clickedPrefs[i]){
                let pref = this.state.preferences[i]
                let category = pref.category
                if (!values.includes(category)){
                    clickedPrefs[i] = !clickedPrefs[i]
                }
            }
        }

        for(var j = 0; j < values.length; j++){
            let category = values[j];

            for(var z=0; z < this.state.preferences.length; z++){
                let preference = this.state.preferences[z];
                if (JSON.stringify(preference.category) === JSON.stringify(category) && clickedPrefs[z] !== true){
                    clickedPrefs[z] = true
                }
            }
        }

        this.setState({ entries: values, clickedPreferences: clickedPrefs}, () => this.determineGetMeetups(this.state.public, values))
    }

    openFormModal = () => {
        this.setState({newMeetupForm: !this.state.newMeetupForm})
    }

    formatCategories = entries => {
        var ids = []
        for (var category in entries){
            ids.push(entries[category].id)
        }
        let categoriesString = ids.join(",")

        return categoriesString.length > 0 ? categoriesString : null
    }

    determineGetMeetups = (isPublic, categories) => {
        if (isPublic) {
            this.props.getMeetups({
                type: "public",
                startDate: this.state.startDate.format("YYYY-MM-DD"),
                endDate: this.state.endDate.format("YYYY-MM-DD"),
                categories: this.formatCategories(categories), 
                coords: {...this.props.user.settings}
            })
        } else {
            this.props.getMeetups({
                type: "private", 
                startDate: this.state.startDate.format("YYYY-MM-DD"),
                endDate: this.state.endDate.format("YYYY-MM-DD"),
                categories: this.formatCategories(categories)
            })
        }
    }

    render(){
        const meetups = this.props.meetups
        
        const renderPreset = () => {
            return (
                <div className={styles.preset}>
                    {this.state.preferences.map((pref, index) => 
                        <div 
                            key={pref.id}
                            onClick={() => this.handlePreferenceClick(index)}
                            className={`${styles.presetCategory} ${this.state.clickedPreferences[index] ? styles.active : ""}`} 
                        >
                            <Avatar variant="square"
                                src={`${process.env.REACT_APP_S3_STATIC_URL}${pref.category.api_label}.png`}
                            />
                            <span>{pref.category.label}</span>
                        </div>
                    )}
                    {this.state.preferences.length === 0 && 
                        <div className="no-entity"> 
                            <ErrorIcon style={{color: "rgb(255, 212, 96)"}}/> 
                        <span className="no-entity-text">No preferences! Add some by pressing the top right pencil.</span>
                    </div>
                    }
                </div>
            )
        }

        return (
            <div className={styles.meetups}>
                <Helmet>
                    <meta charSet="utf-8" />
                    <meta name="description" content="Meetups near you!" />
                    <title>Meetups</title>
                </Helmet>
                <div className={styles.meetupsCategories}>
                    <Paper className={styles.meetupsCategoriesInner} elevation={2}>
                        <div className={styles.meetupsCategoriesTop}>
                            <div>Preferences</div>
                            <div>
                                <Link to={{pathname: `/profile/${this.props.user.id}`, state: {locked: false}}}>
                                    <Tooltip title="Edit Preferences">
                                        <IconButton style={{color: "black"}} edge="end">
                                            <EditIcon/>
                                        </IconButton>
                                    </Tooltip>
                                </Link>
                            </div>
                        </div>
                        {renderPreset()}
                        <div className="search">
                            <DateRangePicker 
                                onDatesChange={this.onDatesChange}
                                onFocusChange={this.onFocusChange}
                                focusedInput={this.state.focusedInput}
                                startDate={this.state.startDate}
                                startDateId="unique_start_date_id"
                                endDate={this.state.endDate}
                                endDateId="unique_end_date_id"
                                keepOpenOnDateSelect
                                minimumNights={0}
                                daySize={50}
                                isOutsideRange={this.state.public ? this.isOutsideRange : () => false}
                                openDirection="up"
                                noBorder
                                showDefaultInputIcon
                                displayFormat="MMM DD"
                                small
                            />
                        </div>
                    </Paper>
                </div>
                <div className={styles.meetupsInnerWrap}>
                    <Paper className={styles.meetupsInnerHeader} elevation={2}>
                        <div>
                            Meetups
                            <Tooltip title="Public Meetups">
                                    <IconButton 
                                        onClick={() => this.handleMeetupsType("public")} 
                                        color={this.state.public ? "primary" :"default"} edge="end"
                                    >
                                        <PublicIcon/>
                                    </IconButton>
                            </Tooltip>
                            <Tooltip title="Your Meetups">
                                <IconButton 
                                    onClick={() => this.handleMeetupsType("private")} 
                                    color={this.state.public ? "default" : "primary"}
                                >
                                    <PeopleIcon/>
                                </IconButton>
                            </Tooltip>
                        </div>   
                        <div className={styles.meetupsSearchBar}>
                            <SearchIcon/>
                            <CategoryAutocomplete 
                                fullWidth={true} size="small" entries={this.state.entries} 
                                handleClick={this.onTagsChange} label="Search Categories..."
                            />
                        </div>
                        <div className={styles.categoryChip}>
                            {this.state.public ? 
                                <>{this.props.user.settings ? this.props.user.settings.radius : "25"} miles</> :
                                <>X miles</>
                            }
                        </div>
                    
                        <Tooltip title="Add Meetup">
                            <IconButton onClick={this.openFormModal} style={{color: "black"}}>
                                <AddIcon/>
                            </IconButton>
                        </Tooltip>
                        <MeetupForm type="create" handleClose={this.openFormModal} open={this.state.newMeetupForm}/>
                    </Paper>
                    
                    <div className={styles.meetupsContainer} style={{minHeight: this.props.isMeetupsFetching ? "calc(100% - 60px)" : "0"}}>
                        {this.props.isMeetupsFetching && <div className="loading" style={{height: "auto"}}><CircularProgress/></div>}
                        {(!this.props.isMeetupsFetching && this.props.isMeetupsInitialized) && 
                            <Grid container spacing={1}>
                                {meetups.map((meetup, i) => 
                                    <Grid key={meetup.id} item xs={12} lg={6} xl={4}>
                                        <Grow in={true} timeout={Math.max((i + 1) * 50, 500)}>
                                            <div className="meetups-cardwrapper">
                                                <MeetupCard key={meetup.id} meetup={meetup}/>
                                            </div>
                                        </Grow>
                                    </Grid>
                                )}
                            </Grid>
                        }
                    </div>
                </div>
            </div>
        )
    }
}

Meetups.propTypes = {
    isMeetupsInitialized: PropTypes.bool.isRequired,
    meetups: PropTypes.arrayOf(meetupPropType).isRequired,
    user: userPropType,
    preferences: PropTypes.arrayOf(preferencePropType).isRequired,
    getMeetups: PropTypes.func.isRequired,
    getPreferences: PropTypes.func.isRequired,
    isMeetupsFetching: PropTypes.bool.isRequired
}

function mapStateToProps(state){
    return {
        user: state.user.user,
        meetups: Object.values(state.meetup.meetups),
        preferences: state.user.preferences,
        isMeetupsInitialized: state.meetup.isMeetupsInitialized,
        isMeetupsFetching: state.meetup.isMeetupsFetching
    }
}

const mapDispatchToProps = {
    getMeetups,
    getPreferences
}

export default connect(mapStateToProps, mapDispatchToProps)(Meetups);