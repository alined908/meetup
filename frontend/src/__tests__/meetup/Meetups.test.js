import React from "react";
import { shallow, mount } from "enzyme";
import { UnderlyingMeetups } from "../../components/meetup/Meetups";
import * as mocks from "../../mocks";
import moment from "moment";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import setupStore from "../../setupTests";

const props = {
  user: mocks.user,
  meetups: [mocks.meetup.uri, mocks.privateMeetup.uri2],
  preferences: mocks.preferences,
  isMeetupsFetching: false,
  isMeetupsInitialized: true,
  getMeetups: jest.fn(),
  getPreferences: jest.fn(),
};

describe("Meetups tests", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("renders without crashing", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("handle componentdidmount", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    expect(props.getMeetups.mock.calls.length).toBe(1);
    expect(props.getPreferences.mock.calls.length).toBe(1);
  });

  it("handle componentdidupdate", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    wrapper.setProps({ preferences: mocks.preferences });
    expect(wrapper.state("preferences")).toBe(mocks.preferences);
  });

  it("should show preferences", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    wrapper.setProps({ preferences: mocks.preferences });
    const preferences = wrapper.find(".presetCategory");
    expect(preferences).toHaveLength(1);
    expect(preferences.text()).toContain("Dessert");
  });

  it("should show public meetups", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    const publicMeetupButton = wrapper.find('[aria-label="public-meetups"]');
    publicMeetupButton.simulate("click");
    expect(wrapper.state("public")).toBe(true);
    expect(props.getMeetups.mock.calls.length).toBe(2);
  });

  it("should show private meetups", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    const privateMeetupButton = wrapper.find('[aria-label="private-meetups"]');
    privateMeetupButton.simulate("click");
    expect(wrapper.state("public")).toBe(false);
    expect(props.getMeetups.mock.calls.length).toBe(2);
  });

  it("should call for new meetups if preferences clicked", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    wrapper.setProps({ preferences: mocks.preferences });
    const preferences = wrapper.find(".presetCategory");
    expect(preferences).toHaveLength(1);
    preferences.simulate("click");
    expect(props.getMeetups.mock.calls.length).toBe(2);
  });

  it("opens form modal onclick", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    expect(wrapper.state("newMeetupForm")).toBe(false);
    const addMeetupButton = wrapper.find('[aria-label="add-meetup"]');
    addMeetupButton.simulate("click");
    expect(wrapper.state("newMeetupForm")).toBe(true);
  });

  it("shows loading state if meetup is fetching", () => {
    const newProps = {
      ...props,
      isMeetupsFetching: true,
      isMeetupsInitialized: false,
    };
    const wrapper = shallow(<UnderlyingMeetups {...newProps} />);
    expect(wrapper.find(".loading")).toHaveLength(1);
  });

  it("calls for meetups if date range is different", () => {
    const store = setupStore({ user: { user: { id: 1 } } });
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <UnderlyingMeetups {...props} />
        </Router>
      </Provider>
    );
    const picker = wrapper.find("withStyles(DateRangePicker)");
    expect(picker).toHaveLength(1);
    expect(
      picker
        .instance()
        .props.onDatesChange({ startDate: moment(), endDate: moment() })
    );
    expect(props.getMeetups.mock.calls.length).toBe(2);
  });

  it("calls for get meetups if onTagsChange", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    wrapper.instance().onTagsChange({}, [mocks.category]);
    expect(props.getMeetups.mock.calls.length).toBe(2);
    expect(wrapper.state("entries")).toStrictEqual([mocks.category]);
  });

  it("handle conflicting clickedPrefrences", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    wrapper.setState({ clickedPreferences: [true] });
    wrapper.instance().onTagsChange({}, [mocks.category]);
    expect(wrapper.state("entries")).toStrictEqual([mocks.category]);
  });

  it("rejects date outside of range", () => {
    const wrapper = shallow(<UnderlyingMeetups {...props} />);
    expect(
      wrapper.instance().isOutsideRange(moment().subtract(1, "days"))
    ).toBe(true);
  });
});
