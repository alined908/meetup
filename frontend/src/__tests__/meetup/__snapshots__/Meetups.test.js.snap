// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meetups tests renders without crashing 1`] = `
<div
  className="innerWrap"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <meta
      charSet="utf-8"
    />
    <meta
      content="Meetups near you!"
      name="description"
    />
    <title>
      Meetups
    </title>
  </HelmetWrapper>
  <div
    className="innerLeft"
    style={
      Object {
        "overflow": "auto",
      }
    }
  >
    <div
      className="innerLeftHeader"
    >
      <div>
        Meetups
      </div>
      <WithStyles(ForwardRef(Button))
        aria-label="add-meetup"
        color="primary"
        onClick={[Function]}
        size="small"
        variant="contained"
      >
        Create
      </WithStyles(ForwardRef(Button))>
    </div>
    <div
      className="innerLeftHeaderBlock"
    >
      <div
        className="hr"
      >
        Settings
      </div>
      <div
        className="innerLeftHeaderBlockAction"
      >
        <div
          className="blockActionHeader"
        >
          Type
        </div>
        <div
          className="blockActionContent"
        >
          <div
            className="meetupTypes"
          >
            <div
              aria-label="public-meetups"
              className="meetupType meetupTypeActive elevate-0"
              onClick={[Function]}
            >
              Public
            </div>
            <div
              aria-label="private-meetups"
              className="meetupType  elevate-0"
              onClick={[Function]}
            >
              Private
            </div>
          </div>
        </div>
      </div>
      <div
        className="innerLeftHeaderBlockAction"
      >
        <div
          className="blockActionHeader"
        >
          Dates
        </div>
        <div
          className="blockActionContent calendar"
        >
          <withStyles(DateRangePicker)
            anchorDirection="left"
            appendToBody={false}
            block={false}
            calendarInfoPosition="bottom"
            customArrowIcon={null}
            customCloseIcon={null}
            customInputIcon={null}
            dayPickerNavigationInlineStyles={null}
            daySize={45}
            disableScroll={false}
            disabled={false}
            displayFormat="MMM DD"
            enableOutsideDays={false}
            endDate={"2020-05-18T02:16:35.233Z"}
            endDateId="unique_end_date_id"
            endDatePlaceholderText="End Date"
            firstDayOfWeek={null}
            focusedInput={null}
            hideKeyboardShortcutsPanel={true}
            horizontalMargin={0}
            initialVisibleMonth={null}
            inputIconPosition="before"
            isDayBlocked={[Function]}
            isDayHighlighted={[Function]}
            isOutsideRange={[Function]}
            isRTL={false}
            keepFocusOnInput={false}
            keepOpenOnDateSelect={true}
            minimumNights={0}
            monthFormat="MMMM YYYY"
            navNext={null}
            navPosition="navPositionTop"
            navPrev={null}
            noBorder={true}
            numberOfMonths={2}
            onClose={[Function]}
            onDatesChange={[Function]}
            onFocusChange={[Function]}
            onNextMonthClick={[Function]}
            onPrevMonthClick={[Function]}
            openDirection="down"
            orientation="horizontal"
            phrases={
              Object {
                "calendarLabel": "Calendar",
                "chooseAvailableEndDate": [Function],
                "chooseAvailableStartDate": [Function],
                "clearDates": "Clear Dates",
                "closeDatePicker": "Close",
                "dateIsSelected": [Function],
                "dateIsSelectedAsEndDate": [Function],
                "dateIsSelectedAsStartDate": [Function],
                "dateIsUnavailable": [Function],
                "enterKey": "Enter key",
                "escape": "Escape key",
                "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
                "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
                "homeEnd": "Home and end keys",
                "jumpToNextMonth": "Move forward to switch to the next month.",
                "jumpToPrevMonth": "Move backward to switch to the previous month.",
                "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
                "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
                "keyboardShortcuts": "Keyboard Shortcuts",
                "leftArrowRightArrow": "Right and left arrow keys",
                "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
                "moveFocusByOneMonth": "Switch months.",
                "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
                "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
                "openThisPanel": "Open this panel.",
                "pageUpPageDown": "page up and page down keys",
                "questionMark": "Question mark",
                "returnFocusToInput": "Return to the date input field.",
                "roleDescription": "datepicker",
                "selectFocusedDate": "Select the date in focus.",
                "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
                "upArrowDownArrow": "up and down arrow keys",
              }
            }
            readOnly={false}
            regular={false}
            renderCalendarInfo={null}
            renderDayContents={null}
            renderMonthElement={null}
            renderMonthText={null}
            renderNavNextButton={null}
            renderNavPrevButton={null}
            renderWeekHeaderElement={null}
            reopenPickerOnClearDates={false}
            required={false}
            screenReaderInputMessage=""
            showClearDates={false}
            showDefaultInputIcon={false}
            small={true}
            startDate={"2020-05-11T02:16:35.233Z"}
            startDateId="unique_start_date_id"
            startDatePlaceholderText="Start Date"
            verticalHeight={null}
            verticalSpacing={22}
            weekDayFormat="dd"
            withFullScreenPortal={false}
            withPortal={false}
          />
        </div>
      </div>
      <div
        className="innerLeftHeaderBlockAction"
      >
        <div
          className="blockActionHeader"
        >
          Radius
        </div>
        <div
          className="blockActionContent"
        >
          <WithStyles(ForwardRef(Slider))
            disabled={false}
            marks={
              Array [
                Object {
                  "value": 5,
                },
                Object {
                  "value": 10,
                },
                Object {
                  "value": 15,
                },
                Object {
                  "value": 20,
                },
                Object {
                  "value": 25,
                },
              ]
            }
            max={25}
            min={5}
            onChange={[Function]}
            onChangeCommitted={[Function]}
            step={5}
            value={25}
            valueLabelDisplay="off"
          />
          <div
            className="blockActionChip"
            style={
              Object {
                "marginLeft": "10px",
              }
            }
          >
            25 miles
          </div>
        </div>
      </div>
      <div
        className="innerLeftHeaderBlockAction"
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        <div
          className="blockActionHeader"
        >
          Categories
        </div>
        <div
          className="blockActionContent"
        >
          <div
            className="meetupsSearchBar elevate-0"
          >
            <CategoryAutocomplete
              entries={Array []}
              fullWidth={true}
              handleClick={[Function]}
              label="Search Categories..."
              size="small"
            />
          </div>
        </div>
      </div>
      <div
        className="hr"
      >
        Preferences
      </div>
      <div
        className="preferences"
      >
        <div
          className="preset"
        />
      </div>
    </div>
  </div>
  <div
    className="innerRight"
  >
    <div
      className="innerRightBlock"
    >
      <div
        className="innerRightBlockHeader"
      >
        <div
          className="hr"
        >
          Meetups Near You
        </div>
      </div>
      <div
        className="meetupsContainer"
        style={
          Object {
            "minHeight": "0",
          }
        }
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            key="1"
            lg={6}
            xl={4}
            xs={12}
          >
            <ForwardRef(Grow)
              in={true}
              timeout={500}
            >
              <div
                className="meetups-cardwrapper"
              >
                <MeetupCard
                  key="1"
                  meetup={
                    Object {
                      "categories": Array [
                        Object {
                          "api_label": "desserts",
                          "id": 1,
                          "image": null,
                          "label": "Dessert",
                        },
                      ],
                      "creator": Object {
                        "avatar": null,
                        "email": "daniel@gmail.com",
                        "first_name": "Daniel",
                        "id": 1,
                        "last_name": "Lee",
                      },
                      "date": "2020-05-10",
                      "events": Object {},
                      "id": 1,
                      "isMeetupEventsInitialized": true,
                      "latitude": 34.228754,
                      "location": "location",
                      "longitude": -118.2351192,
                      "members": Object {
                        "1": Object {
                          "admin": true,
                          "ban": false,
                          "id": 1,
                          "user": Object {
                            "avatar": null,
                            "email": "daniel@gmail.com",
                            "first_name": "Daniel",
                            "id": 1,
                            "last_name": "Lee",
                          },
                        },
                        "2": Object {
                          "admin": false,
                          "ban": false,
                          "id": 2,
                          "user": Object {
                            "avatar": null,
                            "email": "example2@gmail.com",
                            "first_name": "Bob",
                            "id": 2,
                            "last_name": "Jim",
                          },
                        },
                      },
                      "name": "Meetup",
                      "notifs": 0,
                      "public": true,
                      "uri": "uri",
                    }
                  }
                />
              </div>
            </ForwardRef(Grow)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            key="2"
            lg={6}
            xl={4}
            xs={12}
          >
            <ForwardRef(Grow)
              in={true}
              timeout={500}
            >
              <div
                className="meetups-cardwrapper"
              >
                <MeetupCard
                  key="2"
                  meetup={
                    Object {
                      "categories": Array [
                        Object {
                          "api_label": "desserts",
                          "id": 1,
                          "image": null,
                          "label": "Dessert",
                        },
                      ],
                      "creator": Object {
                        "avatar": null,
                        "email": "daniel@gmail.com",
                        "first_name": "Daniel",
                        "id": 1,
                        "last_name": "Lee",
                      },
                      "date": "2020-05-10",
                      "events": Object {},
                      "id": 2,
                      "isMeetupEventsInitialized": true,
                      "latitude": 34.228754,
                      "location": "location",
                      "longitude": -118.2351192,
                      "members": Object {
                        "1": Object {
                          "admin": true,
                          "ban": false,
                          "id": 1,
                          "user": Object {
                            "avatar": null,
                            "email": "daniel@gmail.com",
                            "first_name": "Daniel",
                            "id": 1,
                            "last_name": "Lee",
                          },
                        },
                        "2": Object {
                          "admin": false,
                          "ban": false,
                          "id": 2,
                          "user": Object {
                            "avatar": null,
                            "email": "example2@gmail.com",
                            "first_name": "Bob",
                            "id": 2,
                            "last_name": "Jim",
                          },
                        },
                      },
                      "name": "Meetup",
                      "notifs": 0,
                      "public": false,
                      "uri": "uri2",
                    }
                  }
                />
              </div>
            </ForwardRef(Grow)>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </div>
    </div>
    <Connect(ReduxForm)
      handleClose={[Function]}
      open={false}
      type="create"
    />
  </div>
</div>
`;
